modules/object/main.tf

```diff
@@ -36,4 +36,13 @@ resource "aws_s3_object" "this" {
   lifecycle {
     ignore_changes = [object_lock_retain_until_date]
   }
+
+  override_provider {
```

Comment:
Please change it so that the `override_provider` block is used only if necessary (`override_default_tags` should be `true`).

Also, please keep `lifecycle` block as the last one.